content/en/docs/guides/server-graceful-stop.md

```diff
@@ -0,0 +1,84 @@
+---
+title: Gracefully stopping a server
+description: >-
+  Explains how to gracefully stop/shutdown a gRPC server.
+---
+
+### Overview
+
+gRPC servers often need to shut down gracefully, ensuring that in-flight RPCs
+are completed within a reasonable timeframe and new RPCs are no longer
+accepted. "Graceful shutdown function" facilitates this process, allowing the server to
+transition smoothly without abruptly terminating active connections.
+
+When "Graceful shutdown function" is called, the server immediately stops accepting new
+RPCs. In-flight RPCs are allowed to continue until they complete or a specified
+deadline is reached. Once all active RPCs finish or the deadline expires, the
+server shuts down completely.
+
+### How to use Wait-for-Ready
+
+The exact implementation of "Graceful shutdown function" varies depending on the
+programming language you are using. However, the general pattern
+involves:
+
+- Initiating the graceful shutdown process by calling "Graceful shutdown
+  Function" on your gRPC server object. This function blocks until all
+  currently running RPCs completed. This ensures that in-flight requests are
+  allowed to finish processing.
+- Specify a timeout period to limit the time allowed for in-progress RPCs to
+  finish. It's crucial to call "Forceful Shutdown Function" on server object
+  with a timeout before calling "Graceful shutdown function". This acts as a
```

Comment:
These need to be reversed, first graceful, then forceful.

Reply:
Oh this is intentional. So, forceful shutdown functions needs to be scheduled to execute in a separate thread after some time because graceful shutdown is a blocking call. In go we use a timeAfter function which executes the forceful shutdown in a new goroutine after the specified time passes

Reply:
This is very golang specific.  Other languages would initiate a graceful shutdown with a timeout and then if that indicated that it timed out would call a forceful shutdown.  For example, in Java it looks like:
```Java
        server.shutdown();
        try {
          // Wait for RPCs to complete processing
          if (!server.awaitTermination(30, TimeUnit.SECONDS)) {
            // That was plenty of time. Let's cancel the remaining RPCs
            server.shutdownNow();
            // shutdownNow isn't instantaneous, so give a bit of time to clean resources up
            // gracefully. Normally this will be well under a second.
            server.awaitTermination(5, TimeUnit.SECONDS);
          }
        } catch (InterruptedException ex) {
          server.shutdownNow();
        }
```

I'm not sure of the best approach, because for Go it is very important to setup a timer to run the forceful stop on another thread, while for Java the stop and wait are separate commands so can be done in the same thread.

Reply:
I made it more generic. Let me know if this looks good to you.